app/assets/stylesheets/govuk_publishing_components/components/_button.scss

```diff
@@ -22,6 +22,17 @@ $gem-quiet-button-hover-colour: darken($gem-quiet-button-colour, 5%);
   @include responsive-bottom-margin;
 }
 
+.gem-c-button--inline {
```

Comment:
The modifier sounds like a good addition to me, but adding an example (and a test) for this (using the contextual/embed functionality) would make it easier to review in solitude.
```    embed: |
      <button class="govuk-button gem-c-button--inline">Inline button</button>
      <%= component %>
```